contracts/libraries/CappedMath.sol

```diff
@@ -38,13 +38,10 @@ library CappedMath {
      * @dev Multiplies two unsigned integers, returns 2^256 - 1 on overflow.
      */
     function mulCap(uint _a, uint _b) internal pure returns (uint) {
-        // Gas optimization: this is cheaper than requiring '_a' not being zero, but the
-        // benefit is lost if '_b' is also tested.
-        // See: https://github.com/OpenZeppelin/openzeppelin-solidity/pull/522
-        if (_a == 0)
+        uint c = _a * _b;
+        if (c == 0)
```

Comment:
This is incorrect, we can have "c==0" while "a!=0" and "b!=0" when there is an overflow (ex: a=2^255 and b=2). In this case your code would return 0 while the correct return would be UINT_MAX.

Reply:
This includes a critical vulnerability.

Reply:
Understood.